File: src/main/java/com/itranswarp/summer/context/annotation/Component.java
package com.itranswarp.summer.context.annotation;

import java.lang.annotation.*;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Component {

    /**
     * Bean name. Default to simple class name with first-letter-lowercase.
     */
    String value() default "";

}

File: src/main/kotlin/net/liyze/basin/context/annotation/Component.kt
package net.liyze.basin.context.annotation

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class Component(
    /**
     * Bean name. Default to simple class name with first-letter-lowercase.
     */
    val value: String = ""
)
